### PR TITLE
added country_crossing_factor and avoid_country_crossings

### PR DESCRIPTION
**Pull request by gknisely**
@dgearhart @dnesbitt61 @kevinkreiser

### CODE REVIEW

File: src/sif/autocost.cc [hunk 1]
```diff
@@ -13,11 +13,12 @@ namespace sif {
...
-constexpr float kDefaultGateCost         = 30.0f;  // Seconds
-constexpr float kDefaultTollBoothCost    = 15.0f;  // Seconds
-constexpr float kDefaultTollBoothPenalty = 0.0f;  // Seconds
-
+constexpr float kDefaultManeuverPenalty       = 5.0f;   // Seconds
+constexpr float kDefaultGateCost              = 30.0f;  // Seconds
+constexpr float kDefaultTollBoothCost         = 15.0f;  // Seconds
+constexpr float kDefaultTollBoothPenalty      = 0.0f;   // Seconds
+constexpr float kDefaultCountryCrossingFactor = 600.0f; // Seconds
+constexpr bool  kDefaultAvoidCountryCrossings = true;
```
Comment (dnesbitt61) [inline src/sif/autocost.cc]: make this false

File: src/sif/autocost.cc [hunk 2]
```diff
@@ -179,6 +186,12 @@ bool AutoCost::AllowMultiPass() const {
 // not_thru due to hierarchy transitions
 bool AutoCost::Allowed(const baldr::DirectedEdge* edge,
                        const EdgeLabel& pred) const {
+
+  // Avoid country crossings.
+  if (avoid_country_crossings_ && edge->ctry_crossing()) {
+    return false;
+  }
+
```
Comment (dnesbitt61) [inline src/sif/autocost.cc]: this needs to go after the other tests as it is probably the rarest condition

File: src/sif/autocost.cc [hunk 3]
```diff
@@ -13,11 +13,12 @@ namespace sif {
...
-constexpr float kDefaultManeuverPenalty  = 5.0f;   // Seconds
-constexpr float kDefaultGateCost         = 30.0f;  // Seconds
-constexpr float kDefaultTollBoothCost    = 15.0f;  // Seconds
-constexpr float kDefaultTollBoothPenalty = 0.0f;  // Seconds
-
+constexpr float kDefaultManeuverPenalty       = 5.0f;   // Seconds
+constexpr float kDefaultGateCost              = 30.0f;  // Seconds
+constexpr float kDefaultTollBoothCost         = 15.0f;  // Seconds
+constexpr float kDefaultTollBoothPenalty      = 0.0f;   // Seconds
+constexpr float kDefaultCountryCrossingFactor = 600.0f; // Seconds
```
Comment (dnesbitt61) [inline src/sif/autocost.cc]: Factors influence the path and modulate the cost along an edge. Cost and penalty are used for edge transitions. See the description in the interface document. https://github.com/valhalla/demos/blob/master/docs/valhalla_service.md

File: src/sif/autocost.cc [hunk 4]
```diff
@@ -181,10 +191,17 @@ bool AutoCost::Allowed(const baldr::DirectedEdge* edge,
                        const EdgeLabel& pred) const {
   // Check access, Uturn, simple turn restriction, and not_thru edges
```
Comment (gknisely) [inline src/sif/autocost.cc]: @dnesbitt61 updated everything but I notice that you comment says to check for not_thru edges but it is not in the if statement.  Was it deleted and comment not updated?

**Comment by dnesbitt61** [comment]
:shipit

**Pull request merged**